engine/consensus/matching/engine.go

```diff
@@ -360,14 +392,13 @@ func (e *Engine) onReceipt(originID flow.Identifier, receipt *flow.ExecutionRece
 // 	* `engine.DuplicatedEntryError` [sentinel error] if entry already present in mempool and we don't need to process this again
 //	* exception in case something went wrong
 // 	* nil in case of success
-func (e *Engine) storeReceipt(receipt *flow.ExecutionReceipt, head *flow.Header, log *zerolog.Logger) error {
+func (e *Engine) storeReceipt(receipt *flow.ExecutionReceipt, head *flow.Header) (bool, error) {
```

Comment:
as you have changed the behaviour os this method (turned the sentinel into a dedicated return value), please update the goDoc accordingly